gcloud-java-core/src/test/java/com/google/gcloud/BaseSerializationTest.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotSame;
+
+import org.junit.Test;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.io.Serializable;
+
+/**
+ * Base class for serialization tests. To use this class in your tests override the
+ * {@code serializableObjects()} method to return all objects that must be serializable.
+ */
+public abstract class BaseSerializationTest {
+
+  /**
+   * Returns all objects for which correct serialization must be tested.
+   */
+  public abstract Serializable[] serializableObjects();
```

Comment:
> This could be protected, right?
Yes

> do you think we could have an additional protected abstract method called restorableObjects

No, not all modules have restorable objects. To implement the same behavior we should have a non-abstract `restorableObjects` that returns an empty array and a test that passes without doing nothing, I don't like this idea.